Tests/IntegrationTests/PostgresNIOTests.swift

```diff
@@ -341,6 +341,85 @@ final class PostgresNIOTests: XCTestCase {
         XCTAssertEqual(UUID(uuidString: row?[data: "id"].string ?? ""), UUID(uuidString: "123E4567-E89B-12D3-A456-426655440000"))
     }
 
+    func testInt4Range() throws {
+        var conn: PostgresConnection?
+        XCTAssertNoThrow(conn = try PostgresConnection.test(on: eventLoop).wait())
+        defer { XCTAssertNoThrow( try conn?.close().wait() ) }
+        struct Model: Decodable {
+            let range: Range<Int32>
+        }
+        var results: PostgresQueryResult?
+        XCTAssertNoThrow(results = try conn?.query("""
+        SELECT
+            '[\(Int32.min), \(Int32.max))'::int4range AS range
+        """).wait())
+        XCTAssertEqual(results?.count, 1)
+        var row = results?.first?.makeRandomAccess()
+        let expectedRange: Range<Int32> = Int32.min..<Int32.max
+        var expectedRangeBinaryValue = ByteBuffer()
+        expectedRange.encode(into: &expectedRangeBinaryValue, context: .default)
+        var rangeBytes: ByteBuffer = try XCTUnwrap(row?["range"].bytes)
+        let postgresRange: PostgresRange<Int32>? = try PostgresRange<Int32>.init(from: &rangeBytes, type: .int4Range, format: .binary, context: .default)
```

Comment:
can we try to go to to the final Range directly?

Thinking of:
```swift
  let decodedClosedRange = try row.decode(ClosedRange<Int32>.self)
  let decodedOpenRange = try row.decode(Range<Int32>.self)
```

Reply:
Yes that is much better. See new commit https://github.com/vapor/postgres-nio/pull/330/commits/26b59f03b0db71fabc822bc0ee33e8d20c09b00d.